src/main/sensors/gyroanalyse.c

```diff
@@ -1,35 +1,312 @@
+    if (!isDynamicFilterActive()) {
+        return;
+    }
+
+    // if gyro sampling is > 1kHz, accumulate multiple samples
+    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+        fftAcc[axis] += gyroDev->gyroADC[axis];
+    }
+    fftAccCount++;
+
+    // this runs at 1kHz
+    if (fftAccCount == fftSamplingScale) {
+        fftAccCount = 0;
+
+        //calculate mean value of accumulated samples
+        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+            float sample = fftAcc[axis] / fftSamplingScale;
+            sample = biquadFilterApplyDF2(&fftGyroFilter[axis], sample);
+            gyroData[axis][fftIdx] = sample;
+            if (axis == 0)
+                DEBUG_SET(DEBUG_FFT, 2, lrintf(sample * gyroDev->scale));
+            fftAcc[axis] = 0;
+        }
+
+        fftIdx = (fftIdx + 1) % FFT_WINDOW_SIZE;
+    }
+
+    // calculate FFT and update filters
+    gyroDataAnalyseUpdate(notchFilterDyn);
+}
+
+void stage_rfft_f32(arm_rfft_fast_instance_f32 * S, float32_t * p, float32_t * pOut);
+void arm_cfft_radix8by2_f32( arm_cfft_instance_f32 * S, float32_t * p1);
+void arm_cfft_radix8by4_f32( arm_cfft_instance_f32 * S, float32_t * p1);
+void arm_radix8_butterfly_f32(float32_t * pSrc, uint16_t fftLen, const float32_t * pCoef, uint16_t twidCoefModifier);
+void arm_bitreversal_32(uint32_t * pSrc, const uint16_t bitRevLen, const uint16_t * pBitRevTable);
+
+typedef enum {
+    STEP_ARM_CFFT_F32,
+    STEP_BITREVERSAL,
+    STEP_STAGE_RFFT_F32,
+    STEP_ARM_CMPLX_MAG_F32,
+    STEP_CALC_FREQUENCIES,
+    STEP_UPDATE_FILTERS,
+    STEP_HANNING,
+    STEP_COUNT
+} UpdateStep_e;
+
+/*
+ * Analyse last gyro data from the last FFT_WINDOW_SIZE milliseconds
+ */
+void gyroDataAnalyseUpdate(biquadFilter_t *notchFilterDyn) {
+    static int axis = 0;
+    static int step = 0;
+    arm_cfft_instance_f32 * Sint = &(fftInstance.Sint);
+
+    uint32_t startTime = 0;
+    if (debugMode == (DEBUG_FFT_TIME))
+        startTime = micros();
+
+    DEBUG_SET(DEBUG_FFT_TIME, 0, step);
+    switch (step) {
+        case STEP_ARM_CFFT_F32:
+        {
+            switch (FFT_WINDOW_SIZE / 2) {
+            case 16:
+                // 16us
+                arm_cfft_radix8by2_f32(Sint, fftData);
+                break;
+            case 32:
+                // 35us
+                arm_cfft_radix8by4_f32(Sint, fftData);
+                break;
+            case 64:
+                // 70us
+                arm_radix8_butterfly_f32(fftData, FFT_WINDOW_SIZE / 2, Sint->pTwiddle, 1);
+                break;
+            }
+            DEBUG_SET(DEBUG_FFT_TIME, 1, micros() - startTime);
+            break;
+        }
+        case STEP_BITREVERSAL:
+        {
+            // 6us
+            arm_bitreversal_32((uint32_t*) fftData, Sint->bitRevLength, Sint->pBitRevTable);
+            DEBUG_SET(DEBUG_FFT_TIME, 1, micros() - startTime);
+            step++;
+            //break;
```

Comment:
Rather than have a commented out `//break;` statement, which looks like a temporary measure, comment the line: `// fall through`